src/foam/box/Message.js

```diff
@@ -52,6 +56,22 @@ foam.CLASS({
         RemoteException wrapper = new RemoteException();
         wrapper.setId(t.getClass().getName());
         wrapper.setMessage(t.getMessage());
+        
+        // Special case for CapabilityRuntimeException
+        if (
+          "foam.nanos.crunch.CapabilityRuntimeException"
+          .equals(t.getClass().getName())
+        ) {
+          CapabilityRuntimeException tCapability =
+            (CapabilityRuntimeException) t;
+          CapabilityRequiredRemoteException wrapperCapability =
+            new CapabilityRequiredRemoteException();
+          wrapperCapability.setId(t.getClass().getName());
+          wrapperCapability.setMessage(t.getMessage());
+          wrapperCapability.setCapabilityOptions(
+            tCapability.getCapabilities());
+          wrapper = wrapperCapability;
```

Comment:
When I started looking at it, I thought the same thing. I ended up analyzing this section a little more and found there were duplications. However it seems like for now this is a necessary evil.

Consider conversation around https://nanopay.slack.com/archives/CKPPNQC1J/p1590160732013200

Reply:
This doesn't come up often so it may not be documented very well (and indeed there's no `documentation` property on RemoteException).

`RemoteException` does not represent an Exception class, but rather it's FOAM's model for RPC exceptions. It is necessary to convert the Java Exception into a RemoteException before it can be sent back to the client. I wouldn't call this a "necessary evil" so much as "necessary by design".

Reply:
just for clarity - necessary evil - cause it would be nice to have one exception for one functionality(intercept). But also interestingly, yes also - necessary by design - one server exception to one client exception.

Reply:
Nick is working on modelled exceptions which should allow this to be simplified

Reply:
https://nanopay.atlassian.net/browse/NP-1374